### PR TITLE
Add stale bot

### PR DESCRIPTION
**Pull request by grst**
Add an action to label stale issues as discussed in https://github.com/scverse/governance/issues/41

There are two separate actions for "mark stale" and "close" for the following reason: 
 * According to the specifications in https://github.com/scverse/governance/issues/41, we want to mark any issue that is not `pinned` or an `enhancement` to be marked as stale
 * We only want to *close* issues if they are marked as stale and have the `needs info` label. 

afaict, having separate conditions for closing and labeling is not possible with a single stale action.

### CODE REVIEW

File: .github/actions/close-stale.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    steps:
+      - uses: actions/stale@v5
+        with:
+          days-before-issue-stale: 65535 # We don't want to mark issues as stale in this action
+          days-before-close: 14
+          stale-issue-label: stale
+          any-of-labels: "needs info"
+          stale-issue-message: |
+            This issue has been automatically marked as stale because it has not had recent activity.
+            It will be closed if no further activity occurs. Thank you for your contributions!
```
Comment (Zethson) [inline .github/actions/close-stale.yml:18]: Guess we can get rid of the state issue message? It should not mark things as stale anyways. Cleans the workflow a bit.

File: .github/actions/label-stale.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+jobs:
+  stale:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/stale@v5
+        with:
+          days-before-issue-stale: 60
+          days-before-close: 65535 # We don't want to close issues in this action
+          stale-issue-label: stale
+          exempt-issue-labels: pinned,bug,enhancement
```
Comment (Zethson) [inline .github/actions/label-stale.yml:15]: Does this work with pinned? Do all pinned issues get a pinned label?

File: .github/actions/label-stale.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+          stale-issue-message: |
+            This issue has been automatically marked as stale because it has not had recent activity.
+            It will be closed if no further activity occurs. Thank you for your contributions!
```
Comment (Zethson) [inline .github/actions/label-stale.yml:18]: Maybe an explicit "Please comment to keep this issue open"?

File: .github/actions/label-stale.yml (hunk 2 above)
Comment (Zethson) [reply .github/actions/label-stale.yml:15]: Also, should we add another issue label such as "not stale" or whatever to ensure that we don't have to free important long term issues that are not bugs or enhancements from the bot every 2 months? Might never need it, but you never know.

**Review by Zethson — COMMENTED** [review]
Thank you @grst ! Couple of comments...

File: .github/actions/label-stale.yml (hunk 2 above)
Comment (grst) [reply .github/actions/label-stale.yml:15]: actually, don't think the `pinned` label is related to "pinning" an issue. I thought it was just a conventional name for "not stale". 
E.g. nextflow: https://github.com/nextflow-io/nextflow/issues/2702, but I think I also saw the "pinned" label in other places.

File: .github/actions/close-stale.yml (hunk 1 above)
Comment (grst) [reply .github/actions/close-stale.yml:18]: Hmm, does it still close issues then... but worth a try!

Comment (Zethson) [reply .github/actions/close-stale.yml]: If not, at least shorten it to the bare minimum.

**Comment by ivirshup** [comment]
Are you sure you want to exclude `enhancement` and `bug`? I'm thinking something can be a bug report, but not have enough info to reproduce or figure out what's wrong.

**Comment by ivirshup** [comment]
> I was just following the specification

I'd be fine with sticking with the specification for now. But I don't see it mentioning `"bug"`. Really I had more issue with that than `enhancement` 😅

**Comment by grst** [comment]
oh well :sweat_smile: 
So what do we do? Get rid of it altogether, or keep "enhancement"?

**Comment by ivirshup** [comment]
I'd be fine either way.

As an argument for keeping it: I think if it's obvious it's an enhancement, and not a "question" or "won't implement" maybe it's useful to keep. Maybe it's a feature request that needs fleshing out, but is broadly defined.

**Comment by grst** [comment]
kept `enhancement` for now, can still make the bot more "aggressive" later if required.

File: .github/actions/label-stale.yml (hunk 2 above)
Comment (flying-sheep) [reply .github/actions/label-stale.yml]: I always interpreted it as “don’t mark as stale please”, not as having anything to do with pinned issues. But maybe that could be clearer?

**Review by Zethson — APPROVED** [review]
LGTM. I can live with "pinned" although it seems to be a weird community standard.

**Comment by grst** [comment]
Added `on workflow_dispatch`, as this might be useful for testing the dry-runs.

**Comment by grst** [comment]
@ivirshup, let's merge this? 
With the dry-run activated nothing bad can happen (famous last words), and we'll likely anyway need to follow up on this after inspecting the dry run results.

**Comment by ivirshup** [comment]
Sure. Are you expecting "dry run" results soon?

**Comment by grst** [comment]
As soon as I hit the button? 
And oh well - this: https://github.com/scverse/anndata/pull/817

**Pull request merged**